test/check-machines-create

```diff
@@ -1359,17 +1360,31 @@ vnc_password= "{vnc_passwd}"
                     b.click(".pf-c-modal-box__footer button:contains(Create and run)")
                 else:
                     b.click(".pf-c-modal-box__footer button:contains(Create and edit)")
-            init_state = "Creating VM installation" if dialog.create_and_run else "Creating VM"
-            second_state = "Running" if dialog.create_and_run else "Shut off"
-
-            # Getting rid of timeouts and unecessary waiting at VM creation dialog at
-            # https://github.com/cockpit-project/cockpit-machines/pull/699
-            # caused the VM creation process to get faster and sometimes tests might
-            # not catch "Creating VM" state.
-            # Therefore, test only for such states for VM which are created and
-            # STARTED IMMEDIATELLY as that causes "creating VM" stage to take longer time.
-            if dialog.create_and_run:
-                self._assertVmStates(dialog.name, dialog.connection, init_state, second_state)
+
+            final_state = "Running" if dialog.create_and_run else "Shut off"
+            # Test dowloading RHEL image shows the progress of the download
+            if dialog.os_name == TestMachinesCreate.TestCreateConfig.RHEL_5_0:
+                # Check download percantage is shown as a whole number 0-100 followed by '%'
+                if dialog.create_and_run:
+                    b.wait_in_text(f"#vm-{dialog.name}-{dialog.connection}-state", "Downloading")
```

Comment:
This only downloads a tiny fake image from a local test http server, right? So this seems like a race condition, it could be too fast to catch. Perhaps you can introduce some slowness into the mock server, or do a trick similar to what we do with package installations in cockpit's tests: They stop in the middle and wait for some `/tmp/finish-pk` file to appear. The test can then check the progress state, `m.execute("touch /tmp/finish-pk")` and thus go on in a controlled fashion. That way, you can even pin down the percentage more precisely, if you stop at a well-defined 40% or so.

If you find this is too much work, just make sure the download takes at least 3s?

Reply:
That think that approach in package installations is done with some sleep in post install scripts, so the similar approach would be to have some sleep before/after the download?

One way to make the download slower to set some throttle for http.server, e.g. to make the download file limited to only 100kb/s or something like that. But looking at the python's standard http.server, I could not find any option for that.

However, alternatively, to copy the file to wfile stream, I'm using `shutil.copyfileobj` which copies the file by chunks, and allows us to set buffer size. So what if I create a big file, and set small buffer size, to slow down the process of downloading?

Reply:
Right, http.server does not have any fancy options for rate limiting. I recently added a [packagekit cancel test](https://github.com/cockpit-project/cockpit/commit/76db56ea6440ec8) which blocked the download indefinitely, but this trick does not work here -- we can't read from pipes or char devices, as the web server needs to compute/send Content-Length, so this requires plain files. However, I didn't realize that this re-uses the 500 MB test images. So these are far from "tiny", and should take some time to download. @skobyda mentioned ~ 7s, which is more than enough. I was just concerned it would take 0.1s or so. So I'm happy enough with this.

Reply:
As discussed on IRC, let's change the mock buffer size to 1 kB to ensure this is slow enough, then it should be fine.

Reply:
I see that you now have a sleep(3) (which by itself would already be sufficient) *and* a small buffer size. This is almost too much, but if we care we can drop it again in the future.